### ISSUE TITLE
propertyFormatter formats the value too

### ISSUE THREAD

**Issue by detj** [BrandwatchLtd/remimi#13]
As per the following piece of code, if a `propertyFormatter` is passed through, it formats the custom Mixpanel property keys as well as the values.

```javascript
const renameProperties = (object, formatter) => {
    return Object.keys(object).reduce((memo, key) => {
        memo[formatter(key)] = formatter(object[key]);
        return memo;
    }, {})
};
```

Considering this payload

```javascript
const mixpanel = {
  eventName: 'Achievement Unlocked',
  props: { achievementCode: 'lBdj-kec' }
}

dispatch({ type: ACHIEVEMENT_UNLOCKED, meta: { mixpanel } })
```

This would format the keys as well as the actual values. But, I wouldn't want to have the same logic for formatting the keys and the values. In fact, I don't care about formatting the values at all, just the keys because Mixpanel recommends the keys to be humanized.

The correct behaviour would be this?

```javascript
memo[formatter(key)] = object[key];
```

Any reason why the value is formatted as well? Am I missing something here?

**Comment by HHogg** [comment]
Hi @detj, yes you are correct. I think the original thinking behind this was a consisting prefixing and suffixing of properties and values, but it should definitely be configurable to have one and not the other. 

We will introduce a `valueFormatter` option that allows to retain the existing behaviour.

**Comment by HHogg** [comment]
This should be resolved in #14

**Comment by HHogg** [comment]
Ok this is now available in v2.0.0.

**Comment by HHogg** [comment]
Thanks for taking the time to log the issue @detj! 😄

**Comment by detj** [comment]
O Hey! This is awesome! Thanks for taking the time to fix this @HHogg 😄

**Comment by detj** [comment]
@HHogg I upgraded to `remimi@2.0.0`, there another slight issue with the way propertyFormatter is applied to `increment` array. As per the documentation, the propertyFormatter function is expected to return a string value, but in line no https://github.com/BrandwatchLtd/remimi/blob/f46d3f6db4c952073addf0ceb9e25a5153c87a39/src/mixpanelMiddleware.js#L56

```js
     if (increment && increment.length > 0) {
       mixpanel.people.increment(propertyFormatter(increment));
     }
```

the `propertyFormatter` fn is applied to **increment** which is an array, now, even if we as consumer of remimi provide a fn which accepts an array, the propertyFormatter fn still expects a string, and its possible to format an array of strings and return a string. Would need a different approach ?

Let me know if I should create a separate issue for this or re-opening this one is cool ?

**Comment by HHogg** [comment]
Thanks @detj I'll reopen this issue, and hopefully find some time today to take a look.

**Comment by HHogg** [comment]
Hey @detj, looking at the [increment docs](https://mixpanel.com/help/reference/javascript-full-api-reference#mixpanel.people.increment) for Mixpanel the valid inputs looks like an Object or String, and also a second argument of the number to increment (and also decrement by) which this library doesn't currently support. 

So I've proposed #16 which hopefully aligns it more with the spec. Let us know your thoughts.

**Comment by detj** [comment]
@HHogg Yes, that should work. 👍

**Comment by HHogg** [comment]
Awesome this is now available under [v2.0.1](https://github.com/BrandwatchLtd/remimi/releases/tag/v2.0.1). Let us know how it goes :)
